src/spatio_temporal_voxel_grid.cpp

```diff
@@ -47,15 +47,17 @@ namespace volume_grid
 
 /*****************************************************************************/
 SpatioTemporalVoxelGrid::SpatioTemporalVoxelGrid(
```

Comment:
If the `SpatioTemporalVoxelGrid` only needs the clock, we can give it the clock object from the Layer, no? We can then just remove the ` const nav2_util::LifecycleNode::WeakPtr` completely from the grid